include/mbedtls/check_config.h

```diff
@@ -872,6 +872,10 @@
 #error "MBEDTLS_SSL_CLI_C defined, but not all prerequisites"
 #endif
 
+#if defined(MBEDTLS_SSL_ASYNC_PRIVATE) && !defined(MBEDTLS_X509_CRT_PARSE_C)
```

Comment:
That seems reasonable, considering the ASYNC is about using private keys in TLS, and those are only used with certificates, so we need support for certificate parsing at least.

However, dependencies should not just be checked here, but also documented in `mbedtls_config.h` (see the "Requires:" lines in existing entries).